internal/gatewayapi/securitypolicy.go

```diff
@@ -494,6 +495,17 @@ func (t *Translator) buildOIDC(
 		logoutPath = *oidc.LogoutPath
 	}
 
+	nsName := types.NamespacedName{
+		Namespace: policy.GetNamespace(),
+		Name:      policy.GetName(),
+	}
+	h := fnv.New32a()
+	_, err = h.Write([]byte(nsName.String()))
+	if err != nil {
+		return nil, fmt.Errorf("error generating oauth cookie suffix: %w", err)
+	}
+	suffix := strconv.Itoa(int(h.Sum32()))
```

Comment:
Instead of converting to a string with a decimal number, how about converting to a string with a hexidecimal number?

Since the hash function used generates a 32 bit number, this would consistently generate a string with the same length.

```golang
suffix := fmt.Sprintf("%0X", int(h.Sum32))
```